spring-web/src/main/java/org/springframework/http/codec/json/AbstractJackson2Decoder.java

```diff
@@ -138,20 +139,22 @@ public Flux<Object> decode(Publisher<DataBuffer> input, ResolvableType elementTy
 		Flux<TokenBuffer> tokens = Jackson2Tokenizer.tokenize(processed, mapper.getFactory(), mapper,
 				true, forceUseOfBigDecimal, getMaxInMemorySize());
 
-		ObjectReader reader = getObjectReader(mapper, elementType, hints);
-
-		return tokens.handle((tokenBuffer, sink) -> {
-			try {
-				Object value = reader.readValue(tokenBuffer.asParser(mapper));
-				logValue(value, hints);
-				if (value != null) {
-					sink.next(value);
-				}
-			}
-			catch (IOException ex) {
-				sink.error(processException(ex));
-			}
-		});
+		ObjectReader objectReader = getObjectReader(mapper, elementType, hints);
```

Comment:
Suggestion - should we change getObjectReader to be where the "customize" happens? That way there isn't (as much) code change throughout the class - it's just that we're retrieving an object reader (via Flux now) that will be customized.

Reply:
No, because I don't want double customization happening to then deal with which wins.  So the non-reactive customize can be called in the non-reactive paths and the reactive ones can all the Stream one.  But they both can share the same original getObjectReader